### PR TITLE
refactor: Updates the reduced nav search to use newest Vanilla pattern

### PR DESCRIPTION
**Pull request by petesfrench**
**Don't merge**: Until the branch this PR references has been merged

## Done

- Updates the reduced navigation to use the latest vanilla pattern, found here https://github.com/canonical/vanilla-framework/pull/5210
- It moves the search to be all in the 'normal' navigation

## QA

- Open [demo](https://canonical-com-1294.demos.haus/)
- Check the search functionality, both on desktop and mobile screen sizes

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-13043

### CODE REVIEW

File: templates/navigation/navigation.html [hunk 1]
```diff
@@ -75,27 +75,25 @@
...
-              <i class="p-icon--search"></i>
-            </button>
-          </form>
-        </div>
-      {% endif %}
+      <div class="p-navigation__search">
+        <form class="p-search-box">
+          <input type="search"
+                  class="p-search-box__input"
+                  name="q"
```
Comment (akbarkz) [inline templates/navigation/navigation.html:82]: can we give it a more descriptive name?

**Comment by akbarkz** [comment]
Two finding during QA:

1. The word "Canonical" is not aligned with the bubble logo below
Screenshot 2024-07-09 at 16 32 21

2. When clicking on "Menu" button on mobile screen, it navigates to /navigation. It wasn't like this before.

File: templates/navigation/navigation.html (hunk 1 above)
Comment (petesfrench) [reply templates/navigation/navigation.html:82]: Naming a search input "q" is a widely adopted industry standard

**Comment by petesfrench** [comment]
@akbarkz 

1. I flagged that with Bartek and he informed me that is the expected behaviour, [see the conversation](https://github.com/canonical/vanilla-framework/pull/5210#issuecomment-2213783445)
2. That is the fallback link for when JS fails/doesn't load etc. In this case, as this is the sliding nav, the JS doesn't exist. So this is the expected behavior

**Review by akbarkz — APPROVED** [review]

**Pull request merged**